src/App.tsx

```diff
@@ -108,20 +108,32 @@ export function NavItem({
   );
 }
 
-function BasicLayout(): ReactElement {
+function BasicLayout(): Promise<void> | ReactElement {
   const headerLinks = [...useHeaderAuthLinks()];
   const location = useLocation();
   const auth = useSblAuth();
 
-  // Route users experiencing Authentication service issues to the error page
-  if (auth.error && !location.pathname.includes('/500')) {
-    let errorMessage = auth.error.message;
+  if (auth.error) {
+    const errorMessage = auth.error.message;
 
+    // Authentication service down
     if (errorMessage.includes('Failed to fetch')) {
-      errorMessage = 'The authentication service is unreachable.';
+      if (!location.pathname.includes('/500')) {
+        return (
+          <Navigate
+            to='/500'
+            state={{ message: 'The authentication service is unreachable.' }}
+          />
+        );
+      }
+    }
+    // User's session has expired
+    else if (
+      errorMessage.includes("Session doesn't have required client") &&
+      location.pathname !== '/'
+    ) {
+      return auth.onLogout();
```

Comment:
This assumes we want idled users to re-login.  Otherwise, if they just click `Sign in with Login.gov` they do not seem to need to re-enter credentials.  Maybe that changes when actually using Login.gov?

Reply:
I think that's because their token is still valid with that keycloak setup?